GameOfLife/Calculations/GameFieldSize.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GameOfLife
+{
+    class GameFieldSize
+    {
+        // Field size set up  2D array
+        public  T SetFielDimensions<T>(T UserMessage, T InvalidInpuptMessage)
```

Comment:
Do you need to use the generic T type here? You probably already know, that messages will come in string format and you will be returning some sort of int. Also, now, you are tying incoming type with returning type for the generic type, which could lead to problems down the line.